sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -59,6 +59,12 @@ def __init__(self, http_client, properties, ipython_display,
         super(LivySession, self).__init__()
         assert u"kind" in list(properties.keys())
         kind = properties[u"kind"]
+
+        if not should_heartbeat:
+            properties.pop(u"heartbeatTimeoutInSecond")
```

Comment:
My bad! I missed the `get_session_properties` change. Let me address my own comments with this in mind now.

As for "kind" not being in constants file, that's just bad coding on my side. No reason to repeat my mistakes though.